x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -81,10 +82,36 @@ public class ReservedRolesStore implements BiConsumer<Set<String>, ActionListene
         null,
         new String[] { "*" },
         MetadataUtils.DEFAULT_RESERVED_METADATA,
-        Collections.emptyMap()
+        Collections.emptyMap(),
+        TcpTransport.isUntrustedRemoteClusterEnabled()
+            ? new RoleDescriptor.RemoteIndicesPrivileges[] {
+                new RoleDescriptor.RemoteIndicesPrivileges(
+                    RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("all").allowRestrictedIndices(false).build(),
+                    "*"
+                ),
+                new RoleDescriptor.RemoteIndicesPrivileges(
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices("*")
+                        .privileges("monitor", "read", "view_index_metadata", "read_cross_cluster")
+                        .allowRestrictedIndices(true)
+                        .build(),
+                    "*"
+                ) }
+            : null
     );
     private static final Map<String, RoleDescriptor> RESERVED_ROLES = initializeReservedRoles();
 
+    private static RoleDescriptor.RemoteIndicesPrivileges getRemoteIndicesReadPrivileges(String indexPattern) {
```

Comment:
I considered that, but overall I prefer a more "matching" local and remote indices privileges. These role descriptors are retrievable via APIs like GetRole. I think it works better if the two sections are "matching". Today if CCS is performed with API keys, it will send entire role descriptor across the network. It is not ideal. But in the same time, I haven't seen it being a concern. So my preference is to keep it as is. Alternatively, I can merge privileges for these indices for both local and remote privilege sections. That might be the best of both worlds?

Reply:
Good call that we want to keep these consistent. Let's just leave as is. 

Merging the indices section for local privileges would change the response structure for the GetRole API; it isn't a breaking change per se, but it feels unnecessarily big, given that we would do it to work around an internal implementation quirk.